Remove unused MaybeSetAddrName

maflcko opened the pull request:
.

hebasto reviewed (approved):
ACK fa2d049bfe4f1365011e440adaeb096ddd544143.

Indeed, on master (dbcb5742c48fd26f77e500291d7083e12eec741b) the `CNode::addrName` member has been assigned with a non-empty value in the `CNode::CNode` constructor: https://github.com/bitcoin/bitcoin/blob/dbcb5742c48fd26f77e500291d7083e12eec741b/src/net.cpp#L2978

---

Maybe split out a rename commit with s/`addrName`/`m_addr_name`/?

promag reviewed (commented):
Code review ACK fa2d049bfe4f1365011e440adaeb096ddd544143.

jnewbery commented:
Concept ACK.

This was introduced in https://github.com/bitcoin/bitcoin/commit/f9f5cfc50637f2cd1540923caf337e2651ec1625, and it looks like it would never work as intended, since the `CNode` ctor has always set the `addrName` to a non-empty string.

The comment above explains the intention here: `Also store the name we used to connect in that CNode, so that future FindNode() calls to that name catch this early.`. That comment should be removed if we get rid of this code.

Commit:
Remove unused MaybeSetAddrName

This logic is a no-op since it was introduced in commit
f9f5cfc50637f2cd1540923caf337e2651ec1625.

m_addr_name is never initialized to the empty string, because
ToStringIPPort never returns an empty string.

Commit:
Remove unused RecursiveMutex cs_addrName

hebasto reviewed (approved):
ACK fa973e4d2f50e952fed0aeba7a1a6132f9968a7c, rebased, split in two commits and fixed comment since my [previous](https://github.com/bitcoin/bitcoin/pull/22782#pullrequestreview-736528063) review.

jnewbery commented on src/net.cpp:
```diff
@@ -414,14 +414,10 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
-            if (pnode)
-            {
-                pnode->MaybeSetAddrName(std::string(pszDest));
+            if (pnode) {
```
Could be simplified further to:

```diff
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -415,9 +415,7 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
             }
             // It is possible that we already have a connection to the IP/port pszDest resolved to.
             // In that case, drop the connection that was just created.
-            LOCK(cs_vNodes);
-            CNode* pnode = FindNode(static_cast<CService>(addrConnect));
-            if (pnode) {
+            if (FindNode(static_cast<CService>(addrConnect)));
                 LogPrintf("Failed to open new connection, already connected\n");
                 return nullptr;
             }
```

but probably best left for a future PR. Same for the call to `FindNode()` above.

jnewbery commented on src/net.cpp:
```diff
@@ -534,19 +530,13 @@ std::string ConnectionTypeAsString(ConnectionType conn_type)
-    if (addrName.empty()) {
-        addrName = addrNameIn;
-    }
+std::string CNode::GetAddrName() const
```
Consider putting the definition of this simple getter in the header, for inlining, or making the const member public and removing the getter entirely (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#c131-avoid-trivial-getters-and-setters)

jnewbery reviewed (commented):
Code review ACK fa973e4d2f50e952fed0aeba7a1a6132f9968a7c

Couple of suggestions for follow-ups.

Commit:
Remove GetAddrName

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#c131-avoid-trivial-getters-and-setters
